exercise/reveal_states.py

```diff
@@ -148,6 +148,8 @@ def __init__(self, module: Union[CourseModule, ModulePoints], student: Optional[
         self.max_deviation: Optional[DeadlineRuleDeviation] = None
 
     def get_deadline(self) -> Optional[datetime.datetime]:
+        if len(self.exercises) == 0:
```

Comment:
Empty lists are falsy in `if` conditions. We can avoid the `len()` call and comparison to zero by doing `if not self.exercises:`.